packages/ui/src/accounts/hooks/useBalance.ts

```diff
@@ -14,14 +18,42 @@ export const useBalance = (address: Address = ''): Balances | null => {
   const allMyBalances = useMyBalances()
   const myBalances = allMyBalances?.[address]
 
-  const balances = useFirstObservableValue(
-    () => (isDefined(allMyBalances) && address && !myBalances ? api?.derive.balances.all(address) : undefined),
-    [address, !myBalances]
-  )
+  const balances = useFirstObservableValue(() => {
+    return isDefined(allMyBalances) && address && !myBalances ? api?.derive.balances.all(address) : undefined
+  }, [address, !myBalances])
 
   if (myBalances) {
     return myBalances
   }
 
   return balances ? toBalances(balances) : null
 }
+export const useBalances = (addresses: Address[]): Map<string, Balances | undefined> => {
+  const { api } = useApi()
+  const allMyBalances = useMyBalances()
+
+  const [definedBalances, setDefinedBalances] = useState(new Map<string, Balances>())
+  useEffect(() => {
+    addresses.forEach((address) => {
+      if (definedBalances.get(address)) return
```

Comment:
Even smaller detail (and I know it's something I wrote) but:
```suggestion
      if (definedBalances.has(address)) return
```